### DISCUSSION TITLE
fast_dev_run does not execute pl.LightningModule.test_step()

### DISCUSSION THREAD

**Discussion by jonathanking** [Lightning-AI/pytorch-lightning#12168] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule] [answered]
I may be misunderstanding something about the trainer argument `fast_dev_run`. When I provide `fast_dev_run=1` and I add a print statement in my LightningModule's `test_step` function, the print statement does not appear. In addition, I can see a progress bar for my training set and validation set, but no progress bar appears for the test set.

Is `fast_dev_run` actually running n batches of my training set? I have passed in a DataModule to `trainer.fit()` that includes a test_dataloader.

**Comment by ananthsub** [comment, marked as answer]
`fit` only runs training & validation, not testing.

`trainer.test` runs the test_step

**Reply by jonathanking** [reply]
Thanks for your help, that makes sense. I think it's a little unclear since this gets passed to the trainer's `init` function, whether or not you call `trainer.fit()` or `trainer.test()`.

**Reply by ananthsub** [reply]
@jonathanking I agree. I had proposed this change to clarify this: https://github.com/PyTorchLightning/pytorch-lightning/issues/10444

But it was considered too big of a user experience change to proceed
